File: src/main/scala/org/excavator/boot/optaplanner/domain/Timeslot.scala
package org.excavator.boot.optaplanner.domain

import java.time.{DayOfWeek, LocalTime}

case class Timeslot(dayOfWeek: DayOfWeek, startTime: LocalTime, endTime: LocalTime) {
  override def toString() = dayOfWeek + " " + startTime.toString()

}

File: src/main/java/org/excavator/boot/optaplanner/domain/Timeslot.java
package org.excavator.boot.optaplanner.domain;

import org.apache.commons.lang3.builder.ReflectionToStringBuilder;

import java.time.DayOfWeek;
import java.time.LocalTime;

public class Timeslot {
    private DayOfWeek dayOfWeek;

    private LocalTime startTime;

    private LocalTime endTime;

    public Timeslot(DayOfWeek dayOfWeek, LocalTime startTime, LocalTime endTime) {
        this.dayOfWeek = dayOfWeek;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    public DayOfWeek getDayOfWeek() {
        return dayOfWeek;
    }

    public LocalTime getEndTime() {
        return endTime;
    }

    public LocalTime getStartTime() {
        return startTime;
    }

    @Override
    public String toString() {
        return ReflectionToStringBuilder.toString(this);
    }
}
